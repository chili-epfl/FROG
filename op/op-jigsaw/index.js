import op from '..'

object = {globalStructure: {studentIds: [1,2,3,4,5]}, socialStructure: {}}
configData = {roles: ['chef', 'baker']}


test('Works', () => {
  expect(op.operator(configData, object)).toEqual({}))
}
