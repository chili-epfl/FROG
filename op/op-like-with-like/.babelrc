{

  "plugins": ["transform-class-properties", "syntax-flow"],
  "presets": ["react", "stage-0", "es2015"]
}
