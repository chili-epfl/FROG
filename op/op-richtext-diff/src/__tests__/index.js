import op from '../operatorRunner';

const config = {
  toDiff:
    '{"ops":[{"insert":"hello this is pretty "},{"attributes":{"bold":true},"insert":"interesting"},{"insert":"\\n"}]}'
};

const data = {
  activityData: {
    structure: { groupingKey: 'individual' },
    payload: {
      '1': {
        data: {
          cjr9p77dc002z16sefowtupmk: {
            id: 'cjr9p77dc002z16sefowtupmk',
            li: {
              id: 'cjr9p77c3002j16se9q3u6m90',
              liDocument: {
                liType: 'li-richText',
                payload: {
                  text: {
                    ops: [
                      {
                        insert: 'hi this is very pretty',
                        attributes: {
                          author: 'Xg6zatcGuL7NPK6TN'
                        }
                      },
                      {
                        insert: '\n'
                      }
                    ]
                  }
                }
              }
            }
          }
        }
      }
    }
  }
};

test('Basic diff', () => {
  expect(op(config, data)).toMatchSnapshot();
});
