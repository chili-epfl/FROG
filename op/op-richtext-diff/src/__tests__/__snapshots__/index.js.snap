// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic diff 1`] = `
Object {
  "payload": Object {
    "1": Object {
      "data": Object {
        "cjn1": Object {
          "id": "cjn1",
          "ops": Array [
            Object {
              "insert": "h",
            },
            Object {
              "attributes": Object {
                "background": "#cce8cc",
                "color": "#003700",
              },
              "insert": "ello",
            },
            Object {
              "attributes": Object {
                "author": "Xg6zatcGuL7NPK6TN",
                "background": "#e8cccc",
                "color": "#370000",
                "strike": true,
              },
              "insert": "i",
            },
            Object {
              "insert": " this is ",
            },
            Object {
              "attributes": Object {
                "author": "Xg6zatcGuL7NPK6TN",
                "background": "#e8cccc",
                "color": "#370000",
                "strike": true,
              },
              "insert": "very ",
            },
            Object {
              "insert": "pretty",
            },
            Object {
              "attributes": Object {
                "background": "#cce8cc",
                "color": "#003700",
              },
              "insert": " interesting",
            },
            Object {
              "insert": "
",
            },
          ],
        },
      },
    },
  },
  "structure": Object {
    "groupingKey": "individual",
  },
}
`;
